src/kixi/hecuba/process_helpers.clj

```diff
@@ -0,0 +1,86 @@
+(ns kixi.hecuba.process-helpers
+  (:require [clojure.data.csv        :as csv]
+            [clojure.java.io         :as io]
+            [clojure.tools.logging   :as log]
+            [clojure.set             :as set]
+            [amazonica.aws.s3        :as s3]))
+
+
+;; Helper for listing objects in the s3 bucket
+(defn list-objects-paged
+  "Returns all results in a lazy seq taking into account the pagination."
+  [cred options]
+  (let [resp (s3/list-objects cred options)
+        page-data (:object-summaries resp)]
```

Comment:
the keys and the map could be destructured here to make things a bit neater, but I'd like it shipped more than I'd like it tidied.

`[{:keys [object-summaries truncated? next-marker] :as resp} (s3/list-objects cred options)]`